itests/kit/blockminer.go

```diff
@@ -278,8 +276,11 @@ func (bm *BlockMiner) MineBlocks(ctx context.Context, blocktime time.Duration) {
 			default:
 			}
 
+			now := time.Duration(time.Now().UnixNano())
+			delay := blocktime - (now % blocktime)
+
 			select {
-			case <-time.After(blocktime):
+			case <-time.After(delay):
```

Comment:
This is the real fix. I'm now aligning block times to a multiple of the block time so multiple miners will actually produce blocks at the same time.